src/components/DistanceRequest.js

```diff
@@ -197,62 +186,69 @@ function DistanceRequest({transactionID, report, transaction, mapboxAccessToken,
 
-    const content = (
-        <ScrollView contentContainerStyle={styles.flexGrow1}>
-            <View
-                style={styles.distanceRequestContainer(scrollContainerMaxHeight)}
-                onLayout={(event = {}) => setScrollContainerHeight(lodashGet(event, 'nativeEvent.layout.height', 0))}
-            >
-                <ScrollView
-                    onContentSizeChange={(width, height) => setScrollContentHeight(height)}
-                    onScroll={updateGradientVisibility}
-                    scrollEventThrottle={variables.distanceScrollEventThrottle}
-                    ref={scrollViewRef}
-                >
-                    {_.map(waypoints, (waypoint, key) => {
-                        // key is of the form waypoint0, waypoint1, ...
-                        const index = TransactionUtils.getWaypointIndex(key);
-                        let descriptionKey = 'distance.waypointDescription.';
-                        let waypointIcon;
-                        if (index === 0) {
-                            descriptionKey += 'start';
-                            waypointIcon = Expensicons.DotIndicatorUnfilled;
-                        } else if (index === lastWaypointIndex) {
-                            descriptionKey += 'finish';
-                            waypointIcon = Expensicons.Location;
-                        } else {
-                            descriptionKey += 'stop';
-                            waypointIcon = Expensicons.DotIndicator;
-                        }
-
-                        return (
-                            <MenuItemWithTopDescription
-                                description={translate(descriptionKey)}
-                                title={lodashGet(waypoints, [`waypoint${index}`, 'address'], '')}
-                                iconFill={theme.icon}
-                                secondaryIcon={waypointIcon}
-                                secondaryIconFill={theme.icon}
-                                shouldShowRightIcon
-                                onPress={() => navigateToWaypointEditPage(index)}
-                                key={key}
-                            />
-                        );
-                    })}
-                </ScrollView>
-                {shouldShowGradient && (
-                    <LinearGradient
-                        style={[styles.pAbsolute, styles.b0, styles.l0, styles.r0, {height: halfMenuItemHeight}]}
-                        colors={[StyleUtils.getTransparentColor(theme.modalBackground), theme.modalBackground]}
-                    />
-                )}
-                {hasRouteError && (
-                    <DotIndicatorMessage
-                        style={[styles.mh5, styles.mv3]}
-                        messages={ErrorUtils.getLatestErrorField(transaction, 'route')}
-                        type="error"
-                    />
-                )}
-            </View>
+    const updateWaypoints = useCallback(
```

Comment:
Instead of using `useCallback`, can you define `updateWaypoints` outside `DistanceRequest` component and make it take `transactionID`, `waypoints`, and `waypointsList` as parameters?

Reply:
I just prefer not to put so much logic code inside component definitions. I also prefer the code that makes it easy which values and logic are dependent on state values.

But I think we can keep the current code in this case.